bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -267,4 +276,55 @@ public void filter(@Nullable ContainerRequestContext requestContext) throws IOEx
             }
         }
     }
+
+    private boolean isImplicitUserRole(ContainerRequestContext requestContext) {
+        if (implicitUserRole) {
+            return true;
+        }
+        if (trustedNetworks.isEmpty()) {
+            return false;
+        }
+        return trustedNetworks.stream().anyMatch((networkCDIR) -> {
+            var clientIp = getClientIp(requestContext);
+            var isInTrustedNetwork = isIpInSubnet(clientIp, networkCDIR);
+            if (isInTrustedNetwork) {
+                logger.debug("Granted implicit user role to request with ip {} as part of network {}", clientIp,
+                        networkCDIR);
+            }
+            return isInTrustedNetwork;
+        });
+    }
+
+    private String getClientIp(ContainerRequestContext requestContext) {
+        String ipForwarded = requestContext.getHeaderString("x-forwarded-for");
+        String[] ips = ipForwarded == null ? null : ipForwarded.split(",");
+        String client_ip = (ips == null || ips.length == 0) ? null : ips[0];
+        client_ip = (client_ip == null || client_ip.isEmpty()) ? servletRequest.getRemoteAddr() : client_ip;
+        return client_ip;
+    }
```

Comment:
```suggestion
    private String getClientIp(ContainerRequestContext requestContext) {
        String ipForwarded = Objects.requireNonNullElse(requestContext.getHeaderString("x-forwarded-for"), "");
        String clientIp = ipForwarded.split(",")[0];
        return clientIp.isBlank() ? servletRequest.getRemoteAddr() : clientIp;
    }
```

`String.split` return an array with only one element (the string itself) if no match is found, so if you ensure `ipForwarded` is at least an empty string, you can always take the first element, it is either the first part (if there are more than one),  the only part (if there is only one) or an empty string (if the header was not present). This also reduces the code needed later.